src/v/transform/rpc/client.h

```diff
@@ -62,6 +62,10 @@ class client {
 
     ss::future<result<iobuf, cluster::errc>>
     load_wasm_binary(model::offset, model::timeout_clock::duration timeout);
+    ss::future<result<model::transform_offsets_value>>
+      offset_fetch(model::transform_offsets_key);
+    ss::future<cluster::errc> offset_commit(
+      model::transform_offsets_key, model::transform_offsets_value);
```

Comment:
I am a bit worried about the performance of this API, we do not have batching at any of the stages here, every single offset commit will correspond to replicating a batch. Are we going to commit offsets with the delay on client ?